applications/StructuralMechanicsApplication/custom_advanced_constitutive/wrinkling_linear_2d_law.h

```diff
@@ -0,0 +1,423 @@
+        ) override;
+
+    Vector& CalculateValue(
+        Parameters& rParameterValues,
+        const Variable<Vector>& rThisVariable,
+        Vector& rValue
+        ) override;
+
+     Matrix& CalculateValue(
+        Parameters& rParameterValues,
+        const Variable<Matrix>& rThisVariable,
+        Matrix& rValue
+        ) override;
+
+     array_1d<double, 3 > & CalculateValue(
+        Parameters& rParameterValues,
+        const Variable<array_1d<double, 3 > >& rVariable,
+        array_1d<double, 3 > & rValue
+        ) override;
+
+    array_1d<double, 6 > & CalculateValue(
+        Parameters& rParameterValues,
+        const Variable<array_1d<double, 6 > >& rVariable,
+        array_1d<double, 6 > & rValue
+        ) override;
+
+    void SetValue(
+        const Variable<bool>& rThisVariable,
+        const bool& rValue,
+        const ProcessInfo& rCurrentProcessInfo
+        ) override;
+
+    void SetValue(
+        const Variable<int>& rThisVariable,
+        const int& rValue,
+        const ProcessInfo& rCurrentProcessInfo
+        ) override;
+
+     void SetValue(
+        const Variable<double>& rThisVariable,
+        const double& rValue,
+        const ProcessInfo& rCurrentProcessInfo
+        ) override;
+
+    void SetValue(
+        const Variable<Vector >& rThisVariable,
+        const Vector& rValue,
+        const ProcessInfo& rCurrentProcessInfo
+        ) override;
+
+    void SetValue(
+        const Variable<Matrix >& rThisVariable,
+        const Matrix& rValue,
+        const ProcessInfo& rCurrentProcessInfo
+        ) override;
+
+    void SetValue(
+        const Variable<array_1d<double, 3 > >& rThisVariable,
+        const array_1d<double, 3 > & rValue,
+        const ProcessInfo& rCurrentProcessInfo
+        ) override;
+
+    void SetValue(
+        const Variable<array_1d<double, 6 > >& rThisVariable,
+        const array_1d<double, 6 > & rValue,
+        const ProcessInfo& rCurrentProcessInfo
+        ) override;
+
+    void InitializeMaterial(
+        const Properties& rMaterialProperties,
+        const GeometryType& rElementGeometry,
+        const Vector& rShapeFunctionsValues
+        ) override;
+
+    void InitializeSolutionStep(
+        const Properties& rMaterialProperties,
+        const GeometryType& rElementGeometry,
+        const Vector& rShapeFunctionsValues,
+        const ProcessInfo& rCurrentProcessInfo
+        ) override;
+
+    void FinalizeSolutionStep(
+        const Properties& rMaterialProperties,
+        const GeometryType& rElementGeometry,
+        const Vector& rShapeFunctionsValues,
+        const ProcessInfo& rCurrentProcessInfo
+        ) override;
+
+
+    void InitializeNonLinearIteration(
+        const Properties& rMaterialProperties,
+        const GeometryType& rElementGeometry,
+        const Vector& rShapeFunctionsValues,
+        const ProcessInfo& rCurrentProcessInfo
+        ) override;
+
+
+    void FinalizeNonLinearIteration(
```

Comment:
Can you please use only the new Interface of the CLaw?

Reply:
this is just a "wrapper" for any other law. So as long as they use the old functions I also have to provide them? I think :thinking:

Reply:
then we should update the other laws (I thought they are already)
it has been deprecated for a long time

Can you check please which law is not yet updated?

Reply:
@philbucher, of course, you were right. I removed all deprecated functions, they were unnecessary... :penguin: